community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -404,6 +406,17 @@ def sync_slurm():
     for status, nodes in node_statuses.items():
         do_node_update(status, nodes)
 
+def future_reservation_sync(nodeset:object, nodename:str, state:object)-> None:
+    if future_reservation := lookup().future_reservation(nodeset.future_reservation, nodeset.zone_policy_allow):
+        currentTime = datetime.utcnow()
```

Comment:
Don't increase nesting.

Instead of 

```py
def fn();
  if happy_path:
    # 50 lines of code
```

Do 

```
def fn();
  if not happy_path: 
    return
  # 50 lines of code
```